File: 349-intersection-of-two-arrays.go
func intersection(nums1 []int, nums2 []int) []int {
	s1 := makeMap(nums1)
	ret := []int{}
	for ele := range makeMap(nums2) {
		if _, present := s1[ele]; present {
			ret = append(ret, ele)
		}
	}
	return ret
}

func makeMap(nums []int) map[int]bool {
	m := map[int]bool{}
	for _, n := range nums {
		m[n] = true
	}
	return m
}

File: bwu/array/349-intersection-of-two-arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        set1, set2 = set(nums1), set(nums2)
        ret = []
        for n in set1:
            if n in set2:
                ret.append(n)
                
        return ret

        """
        return list(set(nums1) & set(nums2))
        """
